src/commands/epgsql_cmd_copy_from_stdin.erl

```diff
@@ -0,0 +1,103 @@
+%%% @doc Tells server to switch to "COPY-in" mode
+%%%
+%%% See [https://www.postgresql.org/docs/current/sql-copy.html].
+%%% See [https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-COPY].
+%%%
+%%% When `Format' is `text', copy data should then be delivered using Erlang
+%%% <a href="https://erlang.org/doc/apps/stdlib/io_protocol.html">io protocol</a>.
+%%% See {@link file:write/2}, {@link io:put_chars/2}.
+%%% "End-of-data" marker `\.' at the end of TEXT or CSV data stream is not needed.
+%%%
+%%% When `Format' is `{binary, [epgsql_type()]}', recommended way to deliver data is
+%%% {@link epgsql:copy_send_rows/3}. IO-protocol can be used as well, as long as you can
+%%% do proper binary encoding of data tuples (header and trailer are sent automatically),
+%%% see [https://www.postgresql.org/docs/current/sql-copy.html#id-1.9.3.55.9.4.6].
+%%% When you don't know what are the correct type names for your columns, you could try to
+%%% construct equivalent `INSERT' or `SELECT' statement and call {@link epgsql:parse/2} command.
+%%% It will return `#statement{columns = [#column{type = TypeName}]}' with correct type names.
+%%%
+%%% {@link epgsql_cmd_copy_done} should be called in the end.
+%%%
+%%% This command should not be used with command pipelining!
+%%%
+%%% ```
+%%% > SQuery COPY ... FROM STDIN ...
+%%% < CopyInResponse
+%%% > CopyData*            -- implemented in io protocol, not here
+%%% > CopyDone | CopyFail  -- implemented in epgsql_cmd_copy_done
+%%% < CommandComplete      -- implemented in epgsql_cmd_copy_done
+%%% '''
+-module(epgsql_cmd_copy_from_stdin).
+-behaviour(epgsql_command).
+-export([init/1, execute/2, handle_message/4]).
+-export_type([response/0]).
+
+-type response() :: ok | {error, epgsql:query_error()}.
```

Comment:
This type doesn't match the response in `handle_message(?COPY_IN_RESPONSE, ... `.
It should be `{ok, list(text | binary)}` or maybe something even stricter.

As I was looking through the tests though, I also wondered, what is the point of returning a list of `text | binary`? It's confusing from a API's standpoint.
Since the values will always have to be the same, I think it would be best to return just `text | binary`, instead of a list of them.
Additionally, I think it would also be good to return a `csv` case, as they are distinct formats.

Reply:
> It should be {ok, list(text | binary)} or maybe something even stricter.

Good catch

> what is the point of returning a list of text | binary?

Yeah, one reason might be because it says how many columns are expected in the input data. Another reason is because it's possible in the future that they will be different (arguable). And finally, it's because I added all the asserts some time after I was done with definition of return values :smiley:

> good to return a csv case

I'd say, epgsql doesn't see any difference between text and csv (it only appears in SQL query and epgsql normally doesn't look inside SQL queries). But we can, eg, make 3rd argument of `copy_from_stdin` accept `text | csv | {binary, ...}` and use it instead. Not sure if it worth this complication.